### DISCUSSION TITLE
The error message for CURL accessing SSL certificate expiration on MAC OS is: No such file or directory

### DISCUSSION THREAD

**Discussion by EmptyZeroRain** [openssl/openssl#25237] [Q&A]
I did this
I used curl to access an HTTPS website, but due to the SSL certificate of the website expiring, the error message from curl made it difficult for me to understand
> curl https://192.168.1.111:8000/
His error is like this
curl: (35) OpenSSL/3.3.1: error:80000002:system library::No such file or directory
use curl -v https://192.168.1.111:8000/
His suggestion is

Host 192.168.1.111:8000。was resolved.
IPv6: (none)
IPv4: 192.168.1.111
Trying 192.168.1.111:8000 .....
Connected to 192.168.1.111 port 8000
ALPN: curl offers h2,http/1.1
TLSv1.3 (OUT), TLS handshake, Client hello (1):
TLSv1.3 (IN), TLS handshake, Server hello (2):
TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
TLSv1.3 (IN), TLS handshake, Certificate (11):
TLSv1.3 (OUT), TLS alert, certificate expired (557):
OpenSSL/3.3.1: error:80000002:system library::No such file or directory
closing connection #0
curl: (35) OpenSSL/3.3.1: error:80000002:system library::No such file or directory
I expected the following
Prompt SSL certificate is invalid

curl/libcurl version
curl 8.9.1 (x86_64-apple-darwin21.6.0) libcurl/8.9.1 OpenSSL/3.3.1 (SecureTransport) zlib/1.2.11 brotli/1.1.0 zstd/1.5.6 libidn2/2.3.7 libssh2/1.11.0 nghttp2/1.61.0 librtmp/2.3
Release-Date: 2024-07-31
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz MultiSSL NTLM SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd

operating system
Darwin bogon 21.6.0 Darwin Kernel Version 21.6.0: Mon Jun 24 00:56:10 PDT 2024; root:xnu-8020.240.18.709.2~1/RELEASE_X86_64 x86_64

**Comment by EmptyZeroRain** [comment]
The feedback from the curl team is an issue with OpenSSL。
https://github.com/curl/curl/issues/14498

**Comment by nhorman** [comment]
But you do get an error indicating the certificate is invalid:
```
ALPN: curl offers h2,http/1.1
TLSv1.3 (OUT), TLS handshake, Client hello (1):
TLSv1.3 (IN), TLS handshake, Server hello (2):
TLSv1.3 (IN), TLS handshake, Encrypted Extensions (8):
TLSv1.3 (IN), TLS handshake, Certificate (11):
TLSv1.3 (OUT), TLS alert, certificate expired (557):   <======= HERE
OpenSSL/3.3.1: error:80000002:system library::No such file or directory
```

This error:
```
curl: (35) OpenSSL/3.3.1: error:80000002:system library::No such file or directory
```
Is typically indicative of a problem in which homebrew has a library versioning problem, see https://github.com/openssl/openssl/discussions/25172

**Comment by Sashan** [comment]
this seems to be rather a discussion.
